include/seqan3/io/exception.hpp

```diff
@@ -47,6 +51,19 @@ struct parse_error : std::runtime_error
     {}
 };
 
+//!\brief Thrown if there is an io error in low level io operations such as in std::basic_streambuf operations.
+struct io_error : std::ios_base::failure
+{
+    //!\brief Constructor that forwards the exception string.
+    io_error(std::string const & s,
+             std::error_code const & ec) : std::ios_base::failure{s, ec}
+    {}
+
+    //!\brief Constructor that forwards the exception string.
+    explicit io_error(std::string const & s) : io_error{s, std::make_error_code(std::io_errc::stream)}
```

Comment:
How is this error_code generated? Is there a global shared state of the streams? :scream:

Reply:
I am not sure I understand what you mean? The function returns a new error_code (local variable) initialised with the enum value std::io_errc::stream. But I should change it to `std::error_code{std::io_errc::stream}`, which is effectively the same. But there is no shared state or something.

Reply:
Well it's not important.